conans/client/rest/auth_manager.py

```diff
@@ -27,6 +27,8 @@ def get(self, remote):
         creds = getattr(remote, "_creds", None)
         if creds is None:
             user, token, _ = self._localdb.get_login(remote.url)
+            if token is None:
```

Comment:
Sorry I am not sure what you're trying to say. I have to admit that I am no Python expert. However the idea is to set the token to something invalid in order to make Artifactory return a `401` in case there's no token available.

Thinking about that: That would possibly also ask for username / pw for publicly accessible repos :/
Somehow I feel that is not what you'd like to have, is it?

On the other hand: If one does not want to ask for username & pw for those repos, probably they should not call `conan auth` in the first place, should they?

Reply:
Instead of changeing the `auth_manager.py` I now modified now the code of `rest_client_v2.py` line 35 like this:
```diff
+++ self.bearer = "Bearer %s" % (str(token) if token else "unset")
--- self.bearer = "Bearer %s" % str(token) if token else None
```

Also I can reproduce those integration test fails locally.

Can you tell whether this is something you could possibly consider? If yes I'd start looking into fixing those integration tests.